src/auth/authclient.ts

```diff
@@ -343,6 +363,10 @@ export abstract class AuthClient
       target.set('authorization', authorizationHeader);
     }
 
+    if (xGoogAllowedLocs) {
```

Comment:
The name of the method still implies that it only sets `x-goog-user-project` and `authorization` headers. I'd recommend either creating a separate method to set `x-goog-allowed-locations`, or updating the method name.